python_modules/libraries/dagster-components/dagster_components/core/schema/objects.py

```diff
@@ -29,13 +29,10 @@ class OpSpecSchema(ResolvableSchema):
 
 class AssetDepSchema(ResolvableSchema[AssetDep]):
     asset: str
-    partition_mapping: Optional[str] = None
-
-    class FieldResolvers:
-        @field_resolver("asset")
-        @staticmethod
-        def resolve_asset(context: ResolutionContext, schema: "AssetDepSchema") -> AssetKey:
-            return _resolve_asset_key(schema.asset, context)
+    partition_mapping: Annotated[
+        Optional[str],
+        FieldResolver(lambda context, schema: _resolve_asset_key(schema.asset, context)),
```

Comment:
why does this call `_resolve_asset_key`?

Reply:
whoops, fixed